pkg/cache/clusterqueue_test.go

```diff
@@ -715,20 +715,21 @@ func TestClusterQueueUpdateWithAdmissionCheck(t *testing.T) {
 
 func TestDominantResourceShare(t *testing.T) {
 	cases := map[string]struct {
-		cq          ClusterQueue
+		cq          ClusterQueueSnapshot
 		flvResQ     resources.FlavorResourceQuantities
 		wantDRValue int
 		wantDRName  corev1.ResourceName
 	}{
 		"no cohort": {
-			cq: ClusterQueue{
+			cq: ClusterQueueSnapshot{
 				FairWeight: oneQuantity,
 				Usage: resources.FlavorResourceQuantitiesFlat{
 					{Flavor: "default", Resource: corev1.ResourceCPU}: 1_000,
 					{Flavor: "default", Resource: "example.com/gpu"}:  2,
 				}.Unflatten(),
 				ResourceGroups: []ResourceGroup{
 					{
+						CoveredResources: sets.New(corev1.ResourceCPU, "example.com/gpu"),
```

Comment:
The setup for snapshot seems to require extra field(s) which seem redundant. 

Wouldn't it be better to still use `ClusterQueue` in test cases setup and generate the snapshots based on the ClusterQueues inside `t.Run`?

Reply:
yes, but that can be done in a follow up.